internals/daemon/api_services.go

```diff
@@ -95,10 +95,35 @@ func v1PostServices(c *Command, r *http.Request, _ *UserState) Response {
 		payload.Services = services
 	default:
 		if len(payload.Services) == 0 {
-			return BadRequest("no services to %s provided", payload.Action)
+			return BadRequest("must specify services for %s action", payload.Action)
 		}
 	}
 
+	// TODO (ping @benhoyt) This is clearly the wrong place for this - it's out
```

Comment:
Hmm, yes, this is a tricky one. I don't have the perfect solution off the top of my head, and it's close to EOD, so let's pair tomorrow and discuss an approach.

Reply:
Discussed this and decided that api_services can directly call a Replan function on the check manager - we will keep it simple for now, and if it gets complicated later we could add a replan listener type system then. The API function already has a state lock, so the Replan method needs to not wait for that - this can be mentioned in the documentation for the Replan function (other functions already do this).